crates/vm/src/wasm/memory.rs

```diff
@@ -192,6 +199,26 @@ pub fn write_vp_inputs(
     })
 }
 
+/// Allocated inside the wasm instance using the injected function and return a
+/// ptr to it
+fn wasm_alloc(
+    instance: &wasmer::Instance,
+    store: &mut impl wasmer::AsStoreMut,
+    len: u64,
+) -> Result<u64> {
+    let alloc_fn = instance
+        .exports
+        .get_function(super::run::ALLOC_FN_NAME)
+        .unwrap()
+        .typed::<i32, i32>(&*store)
+        .unwrap();
+
+    let len: i32 = len.try_into().map_err(Error::TryFromInt)?;
```

Comment:
perhaps we can convert len to a `u32`, then cast (using `as`) from `u32` to `i32`, upon calling the guest fn, in order to retain the bit repr. the wasm fn is interpreting the bits as unsigned, so it should be safe